.travis.yml

```diff
@@ -0,0 +1,48 @@
+language: cpp
+dist: trusty
+compiler:
+    - clang
+    - gcc
+env:
+    - LIBRARY_COMBO=gnu-gnu-gnu
+    - LIBRARY_COMBO=ng-gnu-gnu
+    - LIBRARY_COMBO=ng-gnu-gnu BASE_ABI=--disable-mixed-abi
+matrix:
+    exclude:
+        - compiler: gcc
+          env: LIBRARY_COMBO=ng-gnu-gnu
+        - compiler: gcc
+          env: LIBRARY_COMBO=ng-gnu-gnu BASE_ABI=--disable-mixed-abi
+        - compiler: clang
+          env: LIBRARY_COMBO=gnu-gnu-gnu
+sudo: required
+before_install:
+    - sudo apt-get -qq update
+    - sudo apt-get install -y cmake pkg-config libgnutls28-dev libgmp-dev libffi-dev libicu-dev libxml2-dev libxslt1-dev libssl-dev libavahi-client-dev zlib1g-dev
+    - >
+        if [ $LIBRARY_COMBO = 'gnu-gnu-gnu' ];
+        then
+          if [ $CC = 'gcc' ];
+          then
+            sudo apt-get install -y gobjc;
+          fi;
+          sudo apt-get install -y libobjc-4.8-dev;
+        else
+          sudo apt-get install -y libkqueue-dev libpthread-workqueue-dev;
+        fi;
+install: ./travis-deps.sh
+before_script: >
```

Comment:
Happy to merge as-is (rebase please, though. alternatively I'll do it when I get a chance).

Quick check, though, why the distinction between `before_script` and `script`? I tried quickly looking at Travis docs, but I can't find what makes the difference?